sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java

```diff
@@ -1136,19 +1136,37 @@ public Predicate buildCriteriaFilter(ContactCriteria contactCriteria, ContactQue
 			filter = CriteriaBuilderHelper.and(cb, filter, cb.equal(from.get(Case.DELETED), contactCriteria.getDeleted()));
 		}
 
-		if (!DataHelper.isNullOrEmpty(contactCriteria.getPersonUuid())) {
-			// We should allow short and long versions of the UUID so let's do a LIKE behaving like a "starts with"
+		if (!DataHelper.isNullOrEmpty(contactCriteria.getPersonLike())) {
 			Join<Contact, Person> person = joins.getPerson();
-			filter = CriteriaBuilderHelper.and(
+			Join<Person, Location> location = joins.getAddress();
+			Join<Case, Person> casePerson = caze.join(Case.PERSON, JoinType.LEFT);
+			Predicate likeFilters = CriteriaBuilderHelper.buildFreeTextSearchPredicate(
 				cb,
-				filter,
-				CriteriaBuilderHelper.ilikePrecise(cb, person.get(Person.UUID), contactCriteria.getPersonUuid() + "%"));
+				contactCriteria.getPersonLike(),
+				textFilter -> cb.or(
+					// We should allow short and long versions of the UUID so let's do a LIKE behaving like a "starts with"
+					CriteriaBuilderHelper.ilikePrecise(cb, person.get(Person.UUID), textFilter + "%"),
+					CriteriaBuilderHelper.unaccentedIlike(cb, person.get(Person.FIRST_NAME), textFilter),
+					CriteriaBuilderHelper.unaccentedIlike(cb, person.get(Person.LAST_NAME), textFilter),
+					phoneNumberPredicate(
+						cb,
+						(Expression<String>) contactQueryContext.getSubqueryExpression(ContactQueryContext.PERSON_PHONE_SUBQUERY),
+						textFilter),
+					CriteriaBuilderHelper.unaccentedIlike(cb, location.get(Location.CITY), textFilter),
+					CriteriaBuilderHelper.ilike(cb, location.get(Location.POSTAL_CODE), textFilter),
+					// Let's also manage the person of the related case
+					CriteriaBuilderHelper.unaccentedIlike(cb, casePerson.get(Person.UUID), textFilter),
```

Comment:
Move the case person stuff (so case person uuid, case person first and last name, case person phone number) back to the general search field (because they're more associated with the contact than the contact person - it's a bit weird, I know) :)